src/main/scala/io/iohk/ethereum/testmode/TestModeComponentsProvider.scala

```diff
@@ -19,14 +21,42 @@ class TestModeComponentsProvider(
     vm: VMImpl
 ) {
 
-  def ledger(blockchainConfig: BlockchainConfig, sealEngine: SealEngineType): Ledger =
-    new LedgerImpl(
-      blockchain,
-      blockchainConfig,
-      syncConfig,
-      consensus(blockchainConfig, sealEngine),
-      validationExecutionContext
+  private var cache = HashMap.empty[(BlockchainConfig, SealEngineType), Ledger]
+  private var blockQueue = BlockQueue(blockchain, syncConfig)
```

Comment:
Why do we need a cache ? We would like to minimize state if possible. I think there is always only one ledger at a given time anyway during tests right ?

I guess we want to keep the same BlockQueue between the calls. Initially I tried to put all the state in the same place in `TestService`. It's not really a perfect place but it avoids scattering the state everywhere and be able to simplify it when possible. To be consistent we could inject `blockQueue` in `def ledger`, or put the current `Ledger`  in `TestService` ?

Reply:
In my test case, I got two different ledger instances due to different configs.
But maybe I should investigate deeper into this and make sure that for a given test, only one blockchain config is used, do you think it's a worthwhile effort ?
If there is only one config, one can drop the cache.

Reply:
Were you able to investigate this?

Reply:
No, I didn't push further the idea. I don't know if the suite is actually using different config depending on tests, which would prevent a unique one (initial setup + each test config) convergence, and require the current multi-config cache.